apis/database.js

```diff
@@ -80,9 +112,35 @@ export async function addLocalLocation(locationData) {
 
   const createdAt = moment.utc(locationObject.created).toISOString();
   await executeTransaction(
-    'insert into locations (key, coordinateKey, createdAt, resources, status, uploaded) values (?, ?, ?, ?, ?, ?)',
-    [key, key, createdAt, resourcesString, status, 0]
+    'insert into locations (key, coordinateKey, createdAt, resources, status, uploaded, updated) values (?, ?, ?, ?, ?, ?, ?)',
+    [key, key, createdAt, resourcesString, status, 0, 0]
   );
 
   return locationObject;
 }
+
+export async function createOrUpdateDatabase() {
+  const version = await SecureStore.getItemAsync(DATABASE_VERSION_KEY);
+
+  await createDatabases();
+
+  if (version === Constants.manifest.extra.localDatabaseVersion) {
+    return Promise.resolve();
+  }
+
+  const locations = await fetchLocalLocations();
+
+  await clearDatabase();
+  await createDatabases();
+
+  const transactions = locations.map((location) => {
+    const { key, createdAt, resources, status, uploaded = 0, updated = 0 } = location;
+
+    return executeTransaction(
```

Comment:
This might be a stretch but does sqlite have bulk inserts?

Reply:
Not that I am aware of.